marklogic-data-hub/src/main/resources/ml-modules/root/marklogic-data-hub-spark-connector/writeRecords.sjs

```diff
@@ -32,13 +32,16 @@ const headers = writeLib.buildHeaders(endpointConstants);
 
 const flowUtils = new FlowUtils();
 
+const uriTemplateRegEx = endpointConstants.uritemplate ? new RegExp("{([^}]+)}",'gi') : null;
+
 inputArray.forEach(record => {
-  const uri = writeLib.generateUri(record, endpointConstants);
+  const uri = writeLib.generateUri(uriTemplateRegEx,record, endpointConstants);
   const envelope = flowUtils.makeEnvelope(record, headers, [], consts.JSON);
 
   if (insertOptions.temporalCollection) {
     temporal.documentInsert(insertOptions.temporalCollection, uri, envelope, insertOptions.options);
   } else {
+    insertOptions.options.collections = !insertOptions.options.collections ? insertOptions.options.collections : insertOptions.options.collections.filter(x=>x!==uri);
```

Comment:
Is this for the temporal use case, where we don't want the URI to be included in the list of collections if we're doing a temporal insert? Let's leave that out of this PR, it should be in a separate ticket that includes a test to ensure the URI is discarded.

Reply:
Sorry that was intended for the temporal case but placed under the wrong IF branch

Reply:
ready for cmmit